python/ray/serve/api.py

```diff
@@ -1008,3 +1016,26 @@ def __call__(self, requests):
     """
     f._serve_accept_batch = True
     return f
+
+
+def deployment(app: Union[FastAPI, APIRouter]):
+    """Mark a FastAPI application for Serve.
+
+    Example:
+    >>> app = FastAPI()
+    >>> @serve.deployment(app)
+        @app.get("/")
```

Comment:
API still smells a little funny to me but we can discuss that offline. I will focus on the implementation for the review for now.